lib/logstash/outputs/elasticsearch.rb

```diff
@@ -133,6 +133,11 @@ class LogStash::Outputs::ElasticSearch < LogStash::Outputs::Base
   # a timeout occurs, the request will be retried.
   config :timeout, :validate => :number
 
+  # Set the Elasticsearch errors that you don't want to log. By default we log 
+  # every error. A useful example is when you want to skip all 409 errors
+  # which are `document_already_exists_exception`
+  config :silence_failures, :validate => :array, :default  => [""]
```

Comment:
I believe the type is the name of the exception in snake_case form.

e.g. `document-already_exists_exception`, so maybe

```
config :failure_type_logging_whitelist, :validate => :string, :list => true
```

whether the default should be `nil` or `[]` is unclear to me. If default were `[]` this would mean that we do not log anything by default